File: app/src/main/java/com/lucasurbas/counter/data/repository/di/RepositoryModule.java
package com.lucasurbas.counter.data.repository.di;

import com.lucasurbas.counter.data.repository.CounterRepository;
import com.lucasurbas.counter.data.repository.MemoryCounterRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    CounterRepository provideCounterRepository() {
        MemoryCounterRepository repository = new MemoryCounterRepository();
        repository.initData();
        return repository;
    }

}

File: app/src/main/java/com/lucasurbas/counter/data/repository/di/RepositoryModule.kt
package com.lucasurbas.counter.data.repository.di

import com.lucasurbas.counter.data.repository.CounterRepository
import com.lucasurbas.counter.data.repository.MemoryCounterRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RepositoryModule {

    @Provides
    @Singleton
    internal fun provideCounterRepository(): CounterRepository {
        val repository = MemoryCounterRepository()
        repository.initData()
        return repository
    }

}
